### DISCUSSION TITLE
Buy X, Get Y off other items in discount promotions

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/commerce#2288] [Ideas]

**Comment by chasegiunta** [comment]
@lukeholder Is this feature still slated for 3.0 release?

Ah, 3.X, not 3.0 . My mistake.

**Comment by lukeholder** [comment]
3.x sorry.

**Comment by engram-design** [comment]
This one crops up a lot with clients. Buy 2 item, get the 3rd for 50%, or buy 2 and get the 3rd free. These are just examples, and probably wouldn't be used together - just illustrating (otherwise that'd be chaos).

This can _sort of_ be achieved now with min/max limits, but it doesn't really handle:
- a percentage off **only** the 3rd item
- a 100% discount off **only** the 3rd item

Not only that, but it doesn't roll over to multiples. If I buy 4 I should get the 5th for 50% (or free in the above second example).

Also slightly related https://github.com/craftcms/commerce/issues/1728

**Comment by Nellyaa** [comment]
+1 for this feature. Need this too!
